### ISSUE TITLE
How to  get controlnet Preprocessor and use it?

### ISSUE THREAD

**Issue by sahil1572** [Stability-AI/StableSwarmUI#68]
i can see only canny in the list , of model Preprocessor, 
controlnet  model are already installed in my local comfyui I'm using as backend for the StableSwarmUI

**Comment by mcmonkey4eva** [comment]
Currently the main recommended controlnet preprocessors are available via this comfyui extension: https://github.com/Fannovel16/comfyui_controlnet_aux

**Comment by jvachez** [comment]
Models and preprocessors should be downloaded and installed by StableSwarmUI.

The goal of StableSwarmUI is to automate.

**Comment by mcmonkey4eva** [comment]
Not quite the form of automation swarm is built to facilitate but yes

**Comment by mcastril** [comment]
I followed the instructions in [comfyanonymous.github.io](https://comfyanonymous.github.io/ComfyUI_examples/controlnet/) to download and deploy ControlNet models (tried both the original and fp16 versions) and deployed them in `ComfyUI/models/controlnet directory`

The problem is that the Controlnet -> Model does not get a drop down list like the preprocessor and it's not possible to select any model.

I deployed the preprocessor following the instructions in the previous message and it worked, but not with the models...  And I could see that the models are available in the ComfyUI workflow.

Anyway, I am not sure if that ControlNet models are compatible with SDXL

**Comment by mcmonkey4eva** [comment]
SAI official ControlNet model for SDXL are [here (rank 128)](https://huggingface.co/stabilityai/control-lora/tree/main/control-LoRAs-rank128) or [here (rank 256)](https://huggingface.co/stabilityai/control-lora/tree/main/control-LoRAs-rank256).
There are third-party ones out there as well, [here's a collection of them](https://huggingface.co/lllyasviel/sd_control_collection/tree/main) albeit they are variable in quality.


The primary place to look at ControlNet models is the "ControlNets" tab at the bottom of the UI:
![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/a3dcd42e-e27c-4c7d-8b0c-190d3737c962)

There's a refresh button in that UI if you only just added the models (alternately, when in doubt, especially while Swarm is still in alpha, restart the server to make sure things load).
If they're not showing up, you'll need to go to "Server" at the top -> "Server Configuration" and make sure the path for your controlnets is correct.

Generally if you put models _directly into comfy_ and you have a _default install of Swarm_ you used the wrong folder, you should use Swarm's folder. You only want comfy's folders directly if you have a separately managed comfy instance and you have pointed swarm in settings to use comfy's model dir.

**Comment by mr-arindam** [comment]
nothing is showing in the control net tab, there are no models to select at control net model select pane.
I have attached my server configuration page. I am running trough colab installed on my goggle drive. Help.
![image](https://github.com/Stability-AI/StableSwarmUI/assets/122665205/480a3c0c-0df0-4d83-a5b4-ca075f963566)

**Comment by mcmonkey4eva** [comment]
Have you downloaded any to the relevant folder and restarted or refreshed since downloading them?

**Comment by aarongerber** [comment]
@mcmonkey4eva I think I missed the solution in the comments above. The original question was 'How to get controlnet Preprocessor and use it? Like @l33tx0 I only saw Canny available. I saw your response, "Currently the main recommended controlnet preprocessors are available via this comfyui extension: https://github.com/Fannovel16/comfyui_controlnet_aux"... and that is installed inside the custom_nodes... and I do see those under the ComfyUI > controlnet area, but under Control Net above all that shows is Canny.
